pyemu/mat/mat_handler.py

```diff
@@ -1825,6 +1825,10 @@ def extract(self, row_names=None, col_names=None):
         if row_names is None and col_names is None:
             raise Exception("Matrix.extract() " + "row_names and col_names both None")
         extract = self.get(row_names, col_names, drop=True)
+        i,j = extract.x.shape
+        if i==j:
```

Comment:
I think the way to test if the extracted matrix is diagonal is that if i == j and np.count_nonzero(self.x) == np.count_nonzero(np.diag(self.x)).  I think that last condition would make sure there are no off diagonals (in the unlikely case where there are zeros on the diagonal)?

Reply:
Please see reply comments on the other change as to why I was doing this. In hindsight it was stoopid to try a band-aid solution that worked for my situation in PR. Thanks for your patience.